### PR TITLE
modified the travis.yml script to check for shit mappers.

### PR DESCRIPTION
**Pull request by F-OS**
i modified travis.yml so we do not merge a shitty map.

### CODE REVIEW

**Comment by F-OS** [comment]
the first failed build was because i had a brain fart and forgot a -

**Comment by F-OS** [comment]
damn mouse

**Comment by gbasood** [comment]
This currently hits on `step_x/y = 0` and causes failed builds over non-issues.

**Comment by F-OS** [comment]
@gbasood i will recheck the code and make a new request when and if i can fix it

**Comment by optimumtact** [comment]
why do you even have step_x/y's equaling to zero? just strip them out entirely

**Comment by gbasood** [comment]
Actually yeah we really should just strip them out. No need to make a new PR for this either.

**Comment by F-OS** [comment]
wrote a small sed script to remove all step x and y and i will make a new pr with the removed step x and y

**Comment by PJB3005** [comment]
Whoops

**Comment by optimumtact** [comment]
yall having fun? :laughing:

**Comment by F-OS** [comment]
yep

**Comment by clusterfack** [comment]
I feel like if I don't open and close it now I'll be missing out on the fun

**Comment by F-OS** [comment]
@clusterfack do it don't miss out on the fun

**Comment by clusterfack** [comment]
Abandoned or?

File: .travis.yml [hunk 1]
```diff
@@ -34,6 +34,8 @@ before_script:
   - ./install-byond.sh
 
 script:
+  - shopt -s globstar
+  - (! grep 'step_[xy]' maps/**/*.dmm)
```
Comment (PJB3005) [inline .travis.yml]: Erm I think you want `maps/*.dmm`

**Comment by F-OS** [comment]
uhm is this going to be closed or merged?

**Comment by PJB3005** [comment]
If you would fix the PR sure.

**Comment by clusterfack** [comment]
Make it so it will actually pass travis under the new system and then it will be merged

**Comment by F-OS** [comment]
fixed.

**Comment by optimumtact** [comment]
Why didn't you just sed out alll the step_x and step_y ?

**Comment by PsiOmegaDelta** [comment]
`(! grep 'step_[xy]' maps/**/*.dmm)` is how Bay does it, if that's something you wish to consider, simply because we saw no reason to accept step[xy] = 0 either.

**Comment by F-OS** [comment]
@optimumtact because VG has step_x and step_y equaling to zero in its code and this happens:

maps/metaclub.dmm:"anr" = (/obj/structure/window/reinforced{dir = 8},/obj/machin
ery/computer/smelting{name = "Recycling Furnace Console"; step_x = 0; step_y = 0
; smelter_tag = "recycling_furnace"},/turf/simulated/floor{icon_state = "dark"},
/area/maintenance/disposal)
maps/metaclub.dmm:"dGg" = (/obj/item/weapon/cell{step_x = 0},/turf/simulated/flo
or/plating/airless,/area/AIsattele)
maps/metaclub.dmm:"dZU" = (/obj/machinery/door/window{base_state = "left"; dir =
 1; name = "Sample Preparation Loading"; req_access_txt = "47"},/obj/structure/c
able{d1 = 1; d2 = 2; icon_state = "1-2"},/obj/structure/disposalpipe/segment{dir
 = 4; step_x = 0; step_y = 0},/turf/simulated/floor{dir = 1; icon_state = "white
blue"},/area/research_outpost/sample)
maps/metaclub.dmm:"dZV" = (/obj/structure/window/reinforced{dir = 8},/obj/machin
ery/botany/editor,/obj/structure/disposalpipe/segment,/obj/structure/disposalpip
e/segment{dir = 4; step_x = 0; step_y = 0},/turf/simulated/floor{dir = 8; icon_s
tate = "whitegreen"},/area/research_outpost/xenobot)
maps/metaclub.dmm:"dZW" = (/obj/structure/bed/chair,/obj/structure/disposalpipe/
segment{dir = 4; step_x = 0; step_y = 0},/turf/simulated/floor{icon_state = "whi
te"},/area/research_outpost/xenobot)
maps/metaclub.dmm:"dZX" = (/obj/machinery/atmospherics/pipe/simple/scrubbers/hid
den,/obj/machinery/botany/extractor,/obj/structure/disposalpipe/segment{dir = 4;
 step_x = 0; step_y = 0},/turf/simulated/floor{icon_state = "white"},/area/resea
rch_outpost/xenobot)
maps/metaclub.dmm:"eaG" = (/obj/structure/disposalpipe/segment{dir = 4; step_x =
 0; step_y = 0},/turf/simulated/wall/r_wall,/area/research_outpost/spectro)
maps/metaclub.dmm:"eaH" = (/obj/structure/disposalpipe/segment{dir = 4; step_x =
 0; step_y = 0},/turf/simulated/floor{icon_state = "white"},/area/research_outpo
st/sample)
maps/metaclub.dmm:"ebB" = (/obj/machinery/power/apc{dir = 4; pixel_x = 24},/obj/
structure/cable{d1 = 1; d2 = 8; icon_state = "1-8"},/obj/structure/cable{d2 = 8;
 icon_state = "0-8"},/obj/structure/disposalpipe/segment{dir = 4; step_x = 0; st
ep_y = 0},/turf/simulated/floor{icon_state = "white"},/area/research_outpost/sam
ple)
maps/metaclub.dmm:"eef" = (/obj/machinery/door/airlock/research{name = "Anomalou
s Materials"; req_access_txt = "47"},/obj/machinery/door/firedoor/border_only{di
r = 8; name = "Firelock West"},/obj/structure/cable{d1 = 4; d2 = 8; icon_state =
 "4-8"},/obj/structure/disposalpipe/segment{dir = 4; step_x = 0; step_y = 0},/tu
rf/simulated/floor{icon_state = "white"},/area/research_outpost/anomaly)
maps/metaclub.dmm:"eeg" = (/obj/structure/cable{d1 = 1; d2 = 8; icon_state = "1-
8"},/obj/structure/cable{d1 = 2; d2 = 8; icon_state = "2-8"},/obj/structure/disp
osalpipe/segment{dir = 4; step_x = 0; step_y = 0},/turf/simulated/floor{icon_sta
te = "white"},/area/research_outpost/anomaly)
maps/metaclub.dmm:"eeh" = (/obj/structure/disposalpipe/segment{dir = 4; step_x =
 0; step_y = 0},/turf/simulated/floor{icon_state = "white"},/area/research_outpo
st/anomaly)
maps/ministation.dmm:"Me" = (/obj/structure/table/woodentable,/obj/item/weapon/r
eagent_containers/food/drinks/beer{pixel_x = 5; pixel_y = -2; step_x = 0},/turf/
unsimulated/floor{icon_state = "bar"; dir = 2},/area/syndicate_mothership)
maps/oldstation.dmm:"az" = (/obj/structure/cable{tag = ""; icon_state = "4-8"; s
tep_x = 0; pixel_x = 0; d1 = 4; d2 = 8},/turf/simulated/floor,/area/assembly/sho
wroom)
maps/oldstation.dmm:"cb" = (/obj/structure/cable{tag = ""; icon_state = "4-8"; s
tep_x = 0; pixel_x = 0; d1 = 4; d2 = 8},/turf/simulated/wall,/area/hallway/prima
ry/central)
maps/oldstation.dmm:"fu" = (/obj/structure/cable{tag = ""; icon_state = "4-8"; s
tep_x = 0; pixel_x = 0; d1 = 4; d2 = 8},/turf/simulated/floor,/area/derelict/bri
dge)
maps/oldstation.dmm:"fv" = (/obj/machinery/door/firedoor/border_only{dir = 4},/o
bj/structure/cable{tag = ""; icon_state = "4-8"; step_x = 0; pixel_x = 0; d1 = 4
; d2 = 8},/turf/simulated/floor,/area/hallway/primary/central)
maps/oldstation.dmm:"fw" = (/obj/structure/cable{tag = ""; icon_state = "4-8"; s
tep_x = 0; pixel_x = 0; d1 = 4; d2 = 8},/turf/simulated/floor,/area/hallway/prim
ary/central)
maps/oldstation.dmm:"gv" = (/obj/machinery/door/window/eastleft,/obj/structure/c
able{tag = ""; icon_state = "4-8"; step_x = 0; pixel_x = 0; d1 = 4; d2 = 8},/tur
f/simulated/floor,/area/derelict/bridge)
maps/oldstation.dmm:"hP" = (/obj/structure/cable{tag = ""; icon_state = "4-8"; s
tep_x = 0; pixel_x = 0; d1 = 4; d2 = 8},/turf/simulated/floor,/area/crew_quarter
s/captain)
maps/oldstation.dmm:"jc" = (/obj/structure/table,/obj/item/weapon/storage/firsta
id/toxin,/obj/structure/cable{tag = ""; icon_state = "4-8"; step_x = 0; pixel_x
= 0; d1 = 4; d2 = 8},/turf/simulated/floor,/area/medical/genetics)
maps/oldstation.dmm:"jd" = (/obj/structure/table,/obj/item/weapon/paper_bin,/obj
/item/weapon/storage/firstaid/regular,/obj/structure/cable{tag = ""; icon_state
= "4-8"; step_x = 0; pixel_x = 0; d1 = 4; d2 = 8},/turf/simulated/floor,/area/me
dical/genetics)
maps/oldstation.dmm:"je" = (/obj/structure/cable{tag = ""; icon_state = "4-8"; s
tep_x = 0; pixel_x = 0; d1 = 4; d2 = 8},/turf/simulated/wall,/area/medical/genet
ics)
maps/oldstation.dmm:"jf" = (/obj/structure/cable{tag = ""; icon_state = "4-8"; s
tep_x = 0; pixel_x = 0; d1 = 4; d2 = 8},/turf/simulated/floor,/area/derelict/sto
rage/equipment)
maps/oldstation.dmm:"mC" = (/obj/machinery/door/window/eastleft,/obj/structure/c
able{tag = ""; icon_state = "4-8"; step_x = 0; pixel_x = 0; d1 = 4; d2 = 8},/tur
f/simulated/floor,/area/derelict/storage/equipment)
maps/oldstation.dmm:"mD" = (/obj/machinery/door/airlock{req_access_txt = "7"},/o
bj/structure/cable{tag = ""; icon_state = "4-8"; step_x = 0; pixel_x = 0; d1 = 4
; d2 = 8},/turf/simulated/floor,/area/toxins/lab)
maps/oldstation.dmm:"mE" = (/obj/machinery/door/firedoor/border_only{dir = 4},/o
bj/structure/cable{tag = ""; icon_state = "4-8"; step_x = 0; pixel_x = 0; d1 = 4
; d2 = 8},/turf/simulated/floor,/area/toxins/lab)
maps/oldstation.dmm:"mF" = (/obj/structure/cable{tag = ""; icon_state = "4-8"; s
tep_x = 0; pixel_x = 0; d1 = 4; d2 = 8},/turf/simulated/floor,/area/toxins/lab)
maps/oldstation.dmm:"mG" = (/obj/item/weapon/clipboard,/obj/structure/cable{tag
= ""; icon_state = "4-8"; step_x = 0; pixel_x = 0; d1 = 4; d2 = 8},/turf/simulat
ed/floor,/area/toxins/lab)
maps/oldstation.dmm:"mH" = (/obj/structure/cable{d1 = 2; d2 = 8; icon_state = "2
-8"; tag = ""},/obj/structure/cable{tag = ""; icon_state = "4-8"; step_x = 0; pi
xel_x = 0; d1 = 4; d2 = 8},/turf/simulated/floor,/area/toxins/lab)
maps/oldstation.dmm:"mI" = (/obj/structure/cable{tag = ""; icon_state = "4-8"; s
tep_x = 0; pixel_x = 0; d1 = 4; d2 = 8},/turf/simulated/wall/r_wall,/area/toxins
/lab)
maps/oldstation.dmm:"mJ" = (/obj/structure/cable{tag = ""; icon_state = "4-8"; s
tep_x = 0; pixel_x = 0; d1 = 4; d2 = 8},/turf/simulated/floor,/area/derelict/arr
ival)
maps/oldstation.dmm:"mK" = (/obj/effect/landmark{name = "blobstart"},/obj/struct
ure/cable{tag = ""; icon_state = "4-8"; step_x = 0; pixel_x = 0; d1 = 4; d2 = 8}
,/turf/simulated/floor,/area/derelict/arrival)
maps/oldstation.dmm:"nz" = (/obj/machinery/door/firedoor/border_only{dir = 4},/o
bj/structure/cable{tag = ""; icon_state = "4-8"; step_x = 0; pixel_x = 0; d1 = 4
; d2 = 8},/turf/simulated/floor,/area/derelict/storage/equipment)
maps/oldstation.dmm:"oc" = (/obj/machinery/portable_atmospherics/pump,/obj/struc
ture/cable{tag = ""; icon_state = "4-8"; step_x = 0; pixel_x = 0; d1 = 4; d2 = 8
},/turf/simulated/floor,/area/derelict/storage/equipment)
maps/oldstation.dmm:"od" = (/obj/structure/cable{tag = ""; icon_state = "4-8"; s
tep_x = 0; pixel_x = 0; d1 = 4; d2 = 8},/turf/simulated/wall/r_wall,/area/securi
ty/brig)
maps/oldstation.dmm:"oe" = (/obj/structure/closet/secure_closet/security,/obj/st
ructure/cable{tag = ""; icon_state = "4-8"; step_x = 0; pixel_x = 0; d1 = 4; d2
= 8},/turf/simulated/floor,/area/security/brig)
maps/oldstation.dmm:"of" = (/obj/effect/landmark/start{name = "Security Officer"
},/obj/structure/cable{tag = ""; icon_state = "4-8"; step_x = 0; pixel_x = 0; d1
 = 4; d2 = 8},/turf/simulated/floor,/area/security/brig)
maps/oldstation.dmm:"sU" = (/obj/structure/window/reinforced{dir = 4},/obj/struc
ture/cable{tag = ""; icon_state = "4-8"; step_x = 0; pixel_x = 0; d1 = 4; d2 = 8
},/turf/simulated/floor,/area/derelict/hallway/secondary)
maps/oldstation.dmm:"tQ" = (/obj/structure/cable{tag = ""; icon_state = "4-8"; s
tep_x = 0; pixel_x = 0; d1 = 4; d2 = 8},/turf/simulated/wall/r_wall,/area/dereli
ct/bridge/ai_upload)
maps/rust_test.dmm:"af" = (/obj/machinery/door_control{id = 1; step_y = 32},/tur
f/simulated/floor,/area/engine/control)
maps/taxistation.dmm:"acC" = (/obj/structure/disposalpipe/segment{dir = 4; step_
x = 0; step_y = 0},/obj/machinery/atmospherics/pipe/manifold/supply/hidden{dir =
 8},/turf/simulated/floor{icon_state = "white"},/area/medical/medbay)
maps/taxistation.dmm:"acD" = (/obj/structure/disposalpipe/segment{dir = 4; step_
x = 0; step_y = 0},/obj/machinery/atmospherics/pipe/simple/supply/hidden{dir = 4
},/obj/structure/cable{d1 = 1; d2 = 2; icon_state = "1-2"},/turf/simulated/floor
{icon_state = "white"},/area/medical/medbay)
maps/taxistation.dmm:"acN" = (/obj/structure/disposalpipe/segment{dir = 4; step_
x = 0; step_y = 0},/turf/simulated/floor{icon_state = "white"},/area/medical/med
bay)
maps/taxistation.dmm:"acQ" = (/obj/structure/disposalpipe/segment{dir = 4; step_
x = 0; step_y = 0},/obj/machinery/atmospherics/pipe/simple/supply/hidden{dir = 9
},/obj/effect/landmark/start{name = "Medical Doctor"},/turf/simulated/floor{icon
_state = "white"},/area/medical/medbay)
maps/taxistation.dmm:"acX" = (/obj/structure/disposalpipe/segment{dir = 4; step_
x = 0; step_y = 0},/obj/machinery/atmospherics/pipe/simple/supply/hidden{dir = 4
},/turf/simulated/floor{icon_state = "white"},/area/medical/medbay)
maps/taxistation.dmm:"acY" = (/obj/structure/disposalpipe/segment{dir = 4; step_
x = 0; step_y = 0},/obj/machinery/atmospherics/pipe/simple/scrubbers/hidden{dir
= 10},/obj/structure/cable{d1 = 4; d2 = 8; icon_state = "4-8"; pixel_x = 0},/obj
/machinery/door_control{id_tag = "GeneticsDoor"; name = "Genetics Exit Button";
normaldoorcontrol = 1; pixel_x = -8; pixel_y = 24; range = 3},/obj/machinery/cam
era{c_tag = "Genetics Cloning Room"},/turf/simulated/floor{dir = 1; icon_state =
 "whitepurple"},/area/medical/genetics)
maps/taxistation.dmm:"acZ" = (/obj/machinery/door/airlock/glass_medical{id_tag =
 "GeneticsDoor"; name = "Genetics"; req_access_txt = "5"},/obj/machinery/door/fi
redoor,/obj/structure/disposalpipe/segment{dir = 4; step_x = 0; step_y = 0},/obj
/machinery/atmospherics/pipe/simple/scrubbers/hidden{dir = 4},/obj/structure/cab
le{d1 = 4; d2 = 8; icon_state = "4-8"; pixel_x = 0},/turf/simulated/floor{icon_s
tate = "white"},/area/medical/medbay)
maps/taxistation.dmm:"ade" = (/obj/structure/disposalpipe/segment{dir = 4; step_
x = 0; step_y = 0},/obj/machinery/atmospherics/pipe/manifold/scrubbers/hidden{di
r = 8},/turf/simulated/floor{icon_state = "white"},/area/medical/medbay)
maps/taxistation.dmm:"ads" = (/obj/machinery/door/airlock/glass_medical{id_tag =
 null; name = "Break Room"; req_access_txt = "5"; req_one_access_txt = "0"},/obj
/machinery/door/firedoor,/obj/structure/disposalpipe/segment{dir = 4; step_x = 0
; step_y = 0},/obj/structure/cable{d1 = 4; d2 = 8; icon_state = "4-8"; pixel_x =
 0},/turf/simulated/floor{icon_state = "white"},/area/medical/break_room)
maps/taxistation.dmm:"adt" = (/obj/structure/disposalpipe/segment{dir = 4; step_
x = 0; step_y = 0},/obj/structure/cable{d1 = 4; d2 = 8; icon_state = "4-8"; pixe
l_x = 0},/turf/simulated/floor{icon_state = "white"},/area/medical/medbay)
maps/taxistation.dmm:"adC" = (/obj/structure/disposalpipe/segment{dir = 4; step_
x = 0; step_y = 0},/obj/machinery/atmospherics/pipe/simple/supply/hidden{dir = 9
},/obj/structure/cable{d1 = 4; d2 = 8; icon_state = "4-8"; pixel_x = 0},/turf/si
mulated/floor{icon_state = "whitebluefull"},/area/medical/break_room)
maps/taxistation.dmm:"aen" = (/obj/structure/disposalpipe/segment{dir = 4; step_
x = 0; step_y = 0},/obj/machinery/atmospherics/pipe/simple/supply/hidden{dir = 4
},/turf/simulated/floor{icon_state = "whitebluefull"},/area/medical/break_room)
maps/taxistation.dmm:"aiQ" = (/obj/structure/disposalpipe/segment{dir = 4; step_
x = 0; step_y = 0},/obj/machinery/atmospherics/pipe/manifold/supply/hidden{dir =
 4},/obj/structure/cable{d1 = 2; d2 = 8; icon_state = "2-8"},/turf/simulated/flo
or{icon_state = "white"},/area/medical/genetics)
maps/taxistation.dmm:"aiS" = (/obj/structure/disposalpipe/segment{dir = 4; step_
x = 0; step_y = 0},/obj/machinery/atmospherics/pipe/simple/supply/hidden{dir = 4
},/obj/structure/cable{d1 = 4; d2 = 8; icon_state = "4-8"; pixel_x = 0},/turf/si
mulated/floor{icon_state = "white"},/area/medical/genetics)
maps/taxistation.dmm:"ajw" = (/obj/structure/grille,/obj/structure/window/reinfo
rced{dir = 8},/obj/structure/window/reinforced{dir = 1},/obj/structure/window/re
inforced{dir = 4},/obj/structure/disposalpipe/segment{dir = 4; step_x = 0; step_
y = 0},/obj/structure/window/full/reinforced,/obj/machinery/door/poddoor{density
 = 0; icon_state = "pdoor0"; id_tag = "medpriv1"; name = "Privacy Shutters"; opa
city = 0},/turf/simulated/floor/plating,/area/medical/cmo)
maps/taxistation.dmm:"akh" = (/obj/structure/disposalpipe/segment{dir = 4; step_
x = 0; step_y = 0},/obj/machinery/atmospherics/pipe/simple/scrubbers/hidden{dir
= 4},/obj/machinery/alarm{pixel_y = 23},/turf/simulated/floor{icon_state = "whit
e"},/area/medical/genetics)
maps/taxistation.dmm:"cPg" = (/obj/machinery/atmospherics/unary/vent_scrubber/on
,/obj/structure/table/reinforced,/obj/item/device/flashlight/lamp,/obj/item/weap
on/paper_bin{step_y = 0},/turf/simulated/floor{icon_state = "cult"},/area/lawoff
ice)

**Comment by F-OS** [comment]
@PsiOmegaDelta as i said to optimumtact it does not work

**Comment by PsiOmegaDelta** [comment]
Seeing how step_[xy] = 0 should have no effect, remove them?

**Comment by PJB3005** [comment]
But that requires effort.

**Comment by F-OS** [comment]
@PJB3005 eh, anybody want to fix it, i have this thing so uh...

**Pull request merged**